pdns/dnsdistdist/dnsdist-rules.hh

```diff
@@ -1350,3 +1350,66 @@ private:
   boost::optional<std::string> d_value;
   uint8_t d_type;
 };
+
+class PayloadSizeRule : public DNSRule
+{
+  enum class Comparisons : uint8_t { equal, greater, greaterOrEqual, smaller, smallerOrEqual };
+public:
+  PayloadSizeRule(const std::string& comparison, uint16_t size): d_size(size)
+  {
+    if (comparison == "equal") {
+      d_comparison = Comparisons::equal;
+    }
+    else if (comparison == "greater") {
+      d_comparison = Comparisons::greater;
+    }
+    else if (comparison == "greaterOrEqual") {
```

Comment:
Wondering if `notequal` would be nice to have. Though you could create that with `NotRule`.  Come to think of it, you could maybe reduce the number of primitives and let the user use the Boolean combination rules to make the missing ones.

Reply:
Yeah, my reasoning was that `not equal` is really not something I expect to be useful, and it can be done via `NotRule` if really needed indeed. The other ones seemed more useful.